src/chainparams.cpp

```diff
@@ -272,23 +272,32 @@ class CTestNetParams : public CChainParams {
     }
 };
 
+const std::vector<uint8_t> SIGNET_DEFAULT_CHALLENGE = ParseHex("512103ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430210359ef5021964fe22d6f8e05b2463c9540ce96883fe3b278760f048f5189f2e6c452ae");
+const std::vector<std::string> SIGNET_DEFAULT_SEEDS{
+    "seed.signet.bitcoin.sprovoost.nl.",
+    "178.128.221.177",
+    "v7ajjeirttkbnt32wpy3c6w3emwnfr3fkla7hpxcfokr3ysd3kqtzmqd.onion:38333"
+};
+
+SigNetOptions GetDefaultSignetOptions()
+{
+    return {
+        SIGNET_DEFAULT_CHALLENGE,
+        SIGNET_DEFAULT_SEEDS,
+    };
+}
+
 /**
  * Signet: test network with an additional consensus parameter (see BIP325).
  */
 class SigNetParams : public CChainParams {
 public:
-    explicit SigNetParams(const ArgsManager& args) {
-        std::vector<uint8_t> bin;
-        vSeeds.clear();
-
-        if (!args.IsArgSet("-signetchallenge")) {
-            bin = ParseHex("512103ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430210359ef5021964fe22d6f8e05b2463c9540ce96883fe3b278760f048f5189f2e6c452ae");
-            vSeeds.emplace_back("seed.signet.bitcoin.sprovoost.nl.");
-
-            // Hardcoded nodes can be removed once there are more DNS seeds
-            vSeeds.emplace_back("178.128.221.177");
-            vSeeds.emplace_back("v7ajjeirttkbnt32wpy3c6w3emwnfr3fkla7hpxcfokr3ysd3kqtzmqd.onion:38333");
+    explicit SigNetParams(const SigNetOptions& options)
+    {
+        std::vector<uint8_t> bin{options.challenge};
+        vSeeds = options.seeds;
 
+        if (bin == SIGNET_DEFAULT_CHALLENGE) {
```

Comment:
I think if you wanted exactly the same behaviour as current, it would be better to write:

```c++
    if (options.challenge.empty()) { /* use defaults */ } else { /* use values from options */ }
```

Reply:
Done in 7d7900edeff6af18c3bb2c11475ac26414ed78d2